### PR TITLE
test(contrib/drivers/mariadb): add pagination, error, concurrent, rawtype and sharding tests

### PR DESCRIPTION
**Pull request by lingcoder**
## Summary

- Port 11 pagination tests: Page/Limit/Offset, combined with Where/Order, boundary conditions (page 0, large offset), Count with pagination
- Port 8 error-handling tests: invalid table/field names, syntax errors, duplicate key, connection errors, error wrapping and message verification
- Port 5 concurrency tests: parallel read/write with goroutines and WaitGroup, concurrent transactions, race condition verification
- Port 6 raw-type tests: custom type scanning, time.Time handling, json.RawMessage, sql.NullString/NullInt64, []byte fields
- Port 6 sharding/table-name tests: dynamic table name via Sharding callback, table name with prefix, schema.table format

All tests are structurally identical to the MySQL driver baseline. SQL syntax is standard and shared. Package and import references are adapted for MariaDB.

ref #4689

### CODE REVIEW

File: contrib/drivers/mariadb/mariadb_z_unit_feature_model_sharding_test.go [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+	for _, dbName := range dbs {
+		sql := fmt.Sprintf("CREATE DATABASE IF NOT EXISTS `%s`", dbName)
+		_, err := db.Exec(ctx, sql)
+		t.AssertNil(err)
+
+		// Switch to the database
+		sql = fmt.Sprintf("USE `%s`", dbName)
+		_, err = db.Exec(ctx, sql)
+		t.AssertNil(err)
+
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_model_sharding_test.go:45]: Not adopted. This code is ported from the MySQL baseline test (\) which uses the same \ pattern. Changing this would deviate from the baseline and should be addressed at the MySQL level first if needed.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_pagination_test.go [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+	table := createInitTable()
+	defer dropTable(table)
+
+	gtest.C(t, func(t *gtest.T) {
+		result, count, err := db.Model(table).Where("id > ?", 5).AllAndCount(false)
+		t.AssertNil(err)
+		t.Assert(len(result), 5)
+		t.Assert(count, 5)
+		t.Assert(result[0]["id"], 6)
+	})
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_pagination_test.go:50]: Not adopted. MySQL baseline (\) has the same pattern — \ with WHERE filter and no ORDER BY, asserting \. This test validates WHERE filtering logic, not row ordering.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_pagination_test.go [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+
+	gtest.C(t, func(t *gtest.T) {
+		var users []User
+		var count int
+		err := db.Model(table).Where("id <= ?", 5).ScanAndCount(&users, &count, false)
+		t.AssertNil(err)
+		t.Assert(len(users), 5)
+		t.Assert(count, 5)
+		t.Assert(users[0].Id, 1)
+	})
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_pagination_test.go:233]: Not adopted. MySQL baseline (\) has the same pattern — \ with WHERE filter, no ORDER BY, asserting \.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_raw_type_test.go [hunk 4]
```diff
@@ -0,0 +1,905 @@
...
+	defer dropTable(table)
+
+	gtest.C(t, func(t *gtest.T) {
+		// Attempt to insert invalid value (should fail in strict mode)
+		_, err := db.Model(table).Data(g.Map{
+			"status": "invalid_status",
+		}).Insert()
+		// In strict SQL mode, this should produce an error
+		// In non-strict mode, it might insert empty string
+		t.AssertNE(err, nil)
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_raw_type_test.go:774]: Not adopted. MySQL baseline (`mysql_z_unit_feature_raw_type_test.go:774`) has the identical pattern — `t.AssertNE(err, nil)` with the same strict/non-strict mode comment. GoFrame test environments use strict SQL mode by default.

File: contrib/drivers/mariadb/mariadb_unit_init_test.go [hunk 5]
```diff
@@ -61,6 +62,19 @@ func init() {
...
+		Link:        fmt.Sprintf("mariadb:root:%s@tcp(127.0.0.1:3317)/?loc=Local&parseTime=true", TestDbPass),
+		TranTimeout: time.Second * 3,
+	}
+	gdb.AddConfigNode("nodeinvalid", nodeInvalid)
+	if r, err := gdb.NewByGroup("nodeinvalid"); err != nil {
+		gtest.Error(err)
+	} else {
+		dbInvalid = r
+	}
+	dbInvalid = dbInvalid.Schema(TestSchema1)
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_unit_init_test.go:77]: Not adopted. MySQL baseline (`mysql_z_unit_init_test.go:104-109`) has the exact same pattern. In practice, `gdb.NewByGroup` returns a valid DB handle as long as the config group exists — connection errors only occur at actual query time, so `dbInvalid` will not be nil here.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_pagination_test.go [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+
+	// Insert duplicate nicknames
+	for i := 1; i <= 10; i++ {
+		nickname := "name_" + gconv.String((i-1)/2) // Creates duplicates
+		db.Model(table).Data(g.Map{
+			"id":       i,
+			"passport": "pass_" + gconv.String(i),
+			"password": "pwd",
+			"nickname": nickname,
+		}).Insert()
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_pagination_test.go:172]: Not adopted. MySQL baseline (`mysql_z_unit_feature_pagination_test.go:172`) has the same setup pattern — insert in loop without checking error. If inserts fail, the subsequent assertions will naturally fail with clear context.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_concurrent_test.go [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+		for i := 0; i < concurrency; i++ {
+			go func(id int) {
+				defer wg.Done()
+				_, err := db.Model(table).Insert(g.Map{
+					"passport": fmt.Sprintf("user_%d", id),
+					"password": fmt.Sprintf("pass_%d", id),
+					"nickname": fmt.Sprintf("name_%d", id),
+				})
+				t.AssertNil(err)
+			}(i + 1)
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_concurrent_test.go:38]: Not adopted. MySQL baseline (`mysql_z_unit_feature_concurrent_test.go:38`) uses the same pattern — `t.AssertNil(err)` inside goroutines. If this pattern needs improvement, it should be addressed at the MySQL baseline level first.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_concurrent_test.go [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+					_, _ = db.Model(table).Data(g.Map{
+						"nickname": fmt.Sprintf("concurrent_%d", id),
+					}).Where("id", targetId).Update()
+				}(i)
+			case 2: // Query
+				go func(id int) {
+					defer wg.Done()
+					targetId := (id % TableSize) + 1
+					_, _ = db.Model(table).Where("id", targetId).One()
+				}(i)
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_concurrent_test.go:195]: Not adopted. MySQL baseline (`mysql_z_unit_feature_concurrent_test.go:195`) uses the same `_, _ = ...` pattern for concurrent update/query operations. The test intentionally focuses on verifying database consistency (count check) after concurrent operations, not individual operation success.

File: contrib/drivers/mariadb/mariadb_z_unit_feature_model_sharding_test.go [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+	dbs := []string{TestDbNameSh0, TestDbNameSh1}
+	for _, dbName := range dbs {
+		sql := fmt.Sprintf("DROP DATABASE IF EXISTS `%s`", dbName)
+		_, err := db.Exec(ctx, sql)
+		t.AssertNil(err)
+	}
+}
+
+func Test_Sharding_Basic(t *testing.T) {
+	return
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mariadb/mariadb_z_unit_feature_model_sharding_test.go:73]: Not adopted. MySQL baseline (`mysql_z_unit_feature_model_sharding_test.go:80`) uses the same `return` pattern to disable sharding tests. This is a project convention — `t.Skip` has 0 occurrences across the entire GoFrame codebase. Disabled tests use `return` before `gtest.C` by design.
